fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java

```diff
@@ -111,14 +115,26 @@ private Map<String, String> requestHeaders(HttpServletRequest request) {
     if (request.getHeaderNames() != null) {
       for (Enumeration<String> h = request.getHeaderNames(); h.hasMoreElements(); ) {
         String headerName = h.nextElement();
-        String headerValue = request.getHeader(headerName);
+        String headerValue = getRedactedHeaderValue(headerName, request.getHeader(headerName));
         headers.put(headerName, headerValue);
       }
     }
 
     return headers;
   }
 
+  private String getRedactedHeaderValue(String headerName, String headerValue) {
```

Comment:
Looks good to me!

There's one thing you can consider though. I would move the logic you added to a new class that's only purpose would be to redact headers. We achieve a couple of benefits that way:
1. There's better separation of concerns,
2. It's easier to unit test your logic, because no mocks are required and you do not need to deal with the logic of FiatAccessDeniedExceptionHandler,


Take a look at snippet below:
```
private class HeadersRedactor {

        private static final String SECRET_DATA_HEADER_FORMAT = "X-";
        private static final String AUTHORIZATION_HEADER = "Authorization";
        private static final String SECRET_DATA_VALUE = "**REDACTED**";

        Map<String, String> redact(Map<String, String> originalHeaders) {
            var redactedHeaders = new HashMap<String, String>(originalHeaders.size());
            originalHeaders.forEach((key, value) -> {
                
                String newValue;
                if (key.startsWith(SECRET_DATA_HEADER_FORMAT)) {
                    newValue = SECRET_DATA_VALUE;
                } else {
                    if (key.contains(AUTHORIZATION_HEADER)) {
                        newValue = value.trim().split("\n")[0].split(" ")[0] + " " + SECRET_DATA_VALUE;
                    } else {
                        newValue = value;
                        redactedHeaders.put(key, value);
                    }
                }
                redactedHeaders.put(newValue, value);
            });
            return redactedHeaders;
        }

    }
 ```

Feel free to keep your implementation as it is! I just wanted to share how I would do it. :)